src/services/submissionService.ts

```diff
@@ -377,6 +386,191 @@ function validateIdInput(id: string): Result<unknown> {
+            httpStatus: 400
+        })
+    }
+
+    if (filters.limit && filters.limit < 0) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'limit',
+            errorCode: ErrorCode.MIN_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.offset && filters.offset > 10000) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'offset',
+            errorCode: ErrorCode.MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.offset && filters.offset < 0) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'offset',
+            errorCode: ErrorCode.MIN_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.googleMapsUrl === '' || (filters.googleMapsUrl && !filters.googleMapsUrl.trim())) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'googleMapsUrl',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.googleMapsUrl && filters.googleMapsUrl.length > 1028) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'googleMapsUrl',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.googleMapsUrl && hasScriptTags(filters.googleMapsUrl)) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'googleMapsUrl',
+            errorCode: ErrorCode.SUSPICIOUS_PATTERN_DETECTED,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.healthcareProfessionalName === '' || (filters.healthcareProfessionalName && !filters.healthcareProfessionalName.trim())) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'healthcareProfessionalName',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.healthcareProfessionalName && filters.healthcareProfessionalName.length > 128) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'healthcareProfessionalName',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.healthcareProfessionalName && hasScriptTags(filters.healthcareProfessionalName)) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'healthcareProfessionalName',
+            errorCode: ErrorCode.SUSPICIOUS_PATTERN_DETECTED,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.spokenLanguages) {
+        if (filters.spokenLanguages.length === 0) {
```

Comment:
if you make a sub-function, you can check for the length and "return early" before the else part would run. This can simplify the nested ifs.